numpy/lib/function_base.py

```diff
@@ -3954,6 +4082,175 @@ def _lerp(a, b, t, out=None):
     return lerp_interpolation
 
 
+def _find_weighted_index(sk, qsn, interpolation='linear'):
+    dim = sk.shape # (N, d1, d2,..., dk)
+    Nx = dim[0]
+    if Nx == 1:
+        return np.zeros_like(qsn)
+    
+    _sk = sk.reshape(dim[0], -1) # (N,-1)
+    _qsn = qsn.reshape(qsn.shape[0], -1) # (q,-1)
+    indices = []
+    
+    for j in range(_qsn.shape[1]):
```

Comment:
`i` for the first level, `j` for the second level.

Reply:
why?

Reply:
Common usage?

Reply:
It doesn't matter here.
Also, to iterate over A[i,j], j should be put on the outer loop and A[i,j] is common usage.